src/elements/content-explorer/ContentExplorer.js

```diff
@@ -377,6 +387,38 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Call Box Representations API to fetch thumbnails of all files in collection.
+     * The thumbnailUrls array in state will be populated with urls to thumbnails.
+     * If a thumbnail could not be found, that item's entry in thumbnailUrls will be null.
+     *
+     * @private
+     * @param {Object} collection - item collection object
+     * @param {Boolean|void} triggerNavigationEvent - To trigger navigate event and focus grid
+     * @return {void}
+     */
+    async fetchThumbnailUrls(collection: Collection, dimensions: string): Promise<void> {
+        const { items = [] } = collection;
+        if (!items) {
+            return;
+        }
+
+        const thumbnailArray = Array(items.length).fill(null);
+
+        // using for loop instead of items.forEach() to get desired await functionality
+        // TODO: This is bad. Find a better way to do this. Without using await, all requests except final
+        // one were cancelled.  Need to find a cleaner to make sure no requests get cancelled.
+        for (let i = 0; i < items.length; i += 1) {
+            const item = items[i];
+            // eslint-disable-next-line no-await-in-loop
+            await this.api.getFileAPI().getFileThumbnail(item, dimensions, thumbnailUrl => {
+                thumbnailArray[i] = thumbnailUrl;
+                this.setState({ thumbnailUrls: [...thumbnailArray] });
```

Comment:
Is the separate `thumbnailUrls` data structure necessary? Could we the thumbnail url decorate the existing item in `currentCollection`?